src/legacy_wrappers/mongo_client.js

```diff
@@ -15,15 +15,16 @@ module.exports.makeLegacyMongoClient = function (baseClass) {
           ? options
           : undefined;
       options = typeof options !== 'function' ? options : undefined;
+      const client = new this(url, options);
       return maybeCallback(
-        baseClass.connect(url, options).then(client => client[toLegacy]()),
+        client.connect(callback).then(() => client),
```

Comment:
why is the `.then` necessary?  `MongoClient.connect()` returns `this` already, so connecting already returns the client.

Reply:
It's necessary her to return the correct construction of a potential subclass, it is not in the instance.connect so I can remove it from there